docs/extensions/payment-processors/create.md

```diff
@@ -106,7 +106,7 @@ Best practice right now would be to:
     The method `doCancelRecurring()` was added in CiviCRM 5.25. For older versions of CiviCRM you need to implement `cancelSubscription()`.
 
 !!! note
-    From CiviCRM 5.25 `cancelSubscription()` will receive a `\Civi\Payment\PropertyBag` and *not* an array. This means that if you use any `array_` functions or `empty()` to check parameters your implementation will stop working. Also you will *only* receive a standard set of parameters as defined in `\Civi\Payment\PropertyBag` and will need to update to check for those instead of any you were using.
+    From CiviCRM 5.25 `cancelSubscription()` will receive a `\Civi\Payment\PropertyBag` and *not* an array. This means that if you use any `array_` functions or `empty()` to check parameters your implementation will stop working. Also you will receive a standard set of parameters as defined in `\Civi\Payment\PropertyBag`. The PropertyBag does provide mapping of these standard parameters to legacy parameters but we recommend you switch your usage to the new method. For example the recommended way to access the field civicrm_contribution_recur.processor_id from the propertyBag is $bag->getRecurProcessorID(). However, as of writing Authorize.net is using $params['subscriptionId'] to access the same value and the propertyBag is returning the same value.
```

Comment:
Yes. It is actually only once converting back to an array that you are restricted to the new parameters. However we should not be included a reference to Authorize.net and any example should be specific - the reader has no idea who or when it was written and it means nothing to them.
A link to the list of legacy params (as of a specific version of CiviCRM) is a good idea. And perhaps an example. `processor_id` is possibly not the best but also not a bad example because it had a completely unrelated name (ie. `subscriptionId`).
Something like (in a separate paragraph so the information is clearly separated):

For example if your payment processor currently looks for `$params['subscriptionId']` it will continue to work but should be replaced with `$params->getRecurProcessorID()` and the new standardized property name is `recurProcessorID`.
To check if it has a value you would use `if ($params->has('recurProcessorID')) { .. }`

Reply:
OK - so remove 

> This means that if you use any `array_` functions or `empty()` to check parameters your implementation will stop working. Also you will *only* receive a standard set of parameters as defined in `\Civi\Payment\PropertyBag` and will need to update to check for those instead of any you were using."

And say

>For most processors this will require no change. However, if you are using array_ functions to interact with the $params/PropetyBag they will no longer work as it is not an array. In addition we recommend you switch your code to take advantage of the PropertyBag. For example 

```
$processorID = $params['subscriptionId'];
```
Would become 
```
$processorID = $params->getRecurProcessorID();
```

To check if it has a value you would use if ($params->has('recurProcessorID')) { .. }

Reply:
I updated the PR